File: 7. Week 7 - KPIs and Text/exercise/r/visualization.R
# Visualize KPIs using ggplot2

if (!require(ggplot2)) install.packages("ggplot2")
if (!require(readr)) install.packages("readr")
library(ggplot2)
library(readr)

# We need to make sure the path points to our data file
data_path <- "../../data/kpi_stats.csv"

if (file.exists(data_path)) {
  df <- read_csv(data_path, show_col_types = FALSE)
  
  # Reshape for plotting
  # We want a dataframe like: Label, Value
  # TODO: Create a data frame 'kpis' with columns 'Label', 'Value', 'x', and 'y'
  # Label should be c("Total Runs", "Total Distance", "Longest Run", "Avg Pace")
  # Value should come from df columns (df$TotalRuns, df$TotalDistance, etc.)
  
  # kpis <- data.frame(
  #   Label = c("Total Runs", "Total Distance", "Longest Run", "Avg Pace"),
  #   Value = c(...),
  #   x = c(1, 2, 3, 4),
  #   y = 1
  # )
  
  
  # Plot
  # TODO: Create the KPI visualization
  # Use geom_text() to display the values and labels
  
  # p <- ggplot(kpis, aes(x = x, y = y)) + ... 
    
  
  # Uncomment to save
  # ggsave("kpi_dashboard.png", plot = p, width = 10, height = 3)
  # print("Infographic saved to kpi_dashboard.png")

} else {
  print(paste("Error: File not found at", data_path))
}

File: 7. Week 7 - KPIs and Text/exercise/python/visualization.py
import pandas as pd
import matplotlib.pyplot as plt
import os

def visualize():
    data_path = '../../data/kpi_stats.csv'
    
    if not os.path.exists(data_path):
        print(f"Data not found at {data_path}")
        return

    # TODO: Load data
    # df = ...
    
    # Get the single row of KPIs
    # kpis = df.iloc[0]
    
    # Plot "Infographic"
    fig, ax = plt.subplots(figsize=(12, 4))
    ax.axis('off')
    
    # Define Metrics to show
    metrics = [
        # TODO: Add your metrics here as tuples (Label, Value)
        # ('Total Runs', ...),
        # ...
    ]
    
    # Position them
    for i, (label, value) in enumerate(metrics):
        x = (i + 0.5) / len(metrics)
        # TODO: Add text to the plot using ax.text()
        # ax.text(...) # For value
        # ax.text(...) # For label
        pass # Remove this pass when you add code
        
    plt.title('Quick Global Stats', fontsize=18)
    plt.tight_layout()
    # TODO: Save
    # plt.savefig(...)
    
    print("KPI Infographic saved to kpi_dashboard.png")
    plt.show()

if __name__ == "__main__":
    visualize()
